core/src/main/java/com/linecorp/armeria/common/util/ThreadFactoryBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Function;
+
+/**
+ * Builds a new {@link ThreadFactory} instance.
+ */
+public final class ThreadFactoryBuilder {
+
+    private String threadNamePrefix;
+    private boolean daemon;
+    private int priority = Thread.NORM_PRIORITY;
+    private ThreadGroup threadGroup;
+    private Function<? super Runnable, ? extends Runnable> taskFunction = Function.identity();
+    private ThreadFactoryProvider threadFactoryProvider;
```

Comment:
nit: final
Could you move this up before `daemon` so that final fields stick together?

Reply:
Made it `final` and arranged fields. Thank you.